dissect/target/tools/utils.py

```diff
@@ -196,3 +198,24 @@ def persist_execution_report(output_dir: Path, report_data: Dict, timestamp: dat
     report_full_path = output_dir / report_filename
     report_full_path.write_text(json.dumps(report_data, sort_keys=True, indent=4))
     return report_full_path
+
+
+def catch_sigpipe(func: Callable) -> Callable:
```

Comment:
Not sure if this implementation is correct, what is the practical use case?

According to the docs I found (http://www.pixelbeat.org/programming/sigpipe_handling.html), the CLI result for:

`target-query -l | head -n1`

should be:

`Available plugins:`

(like:  yes 1234 | head -n1)

But the actual output seems to be:

```
Available plugins:
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
BrokenPipeError: [Errno 32] Broken pipe
```
Not sure if that is fixable or if we even want to change that.
So, can you help me understand, what is the practical value of this change?

Reply:
If you want to get rid of those messages, apparently you have to close `sys.stderr` just before exiting https://bugs.python.org/issue11380,#msg198345

The idea was to not have backtraces or extraneous output confusing the user when dealing with an external situation where we know things are "broken" and we know we should stop the whole process anyway.